### ISSUE TITLE
Learning rate scheduler do not work on AllenNLP v2

### ISSUE THREAD

**Issue by bratao** [allenai/allennlp#4955]
Hello, I´m porting my code to the v2 version, and realized that the Learning rate scheduler was not working. 

After inspecting the training code, I realized that on def _try_train(self) the variable:

`this_epoch_val_metric: float = 0.0` never changes it values

However the scheduler API requires a validation metric

```
# The Scheduler API is agnostic to whether your schedule requires a validation metric -
            # if it doesn't, the validation metric passed here is ignored.
            if self._learning_rate_scheduler:
                self._learning_rate_scheduler.step(this_epoch_val_metric)
            if self._momentum_scheduler:
                self._momentum_scheduler.step(this_epoch_val_metric)
```

I guess it is related to _metric_tracker change, that now receives a list of metrics. But I guess @dirkgr would solve it more elegantly as he was the designer of the improved metric tracker

**Comment by wlhgtc** [comment]
Not all  scheduler need metric , you could see code in [here](https://github.com/allenai/allennlp/blob/main/allennlp/training/learning_rate_schedulers/slanted_triangular.py#L97).
Only class based on `_PyTorchLearningRateSchedulerWithMetricsWrapper ` like  [`ReduceOnPlateauLearningRateScheduler `](https://github.com/allenai/allennlp/blob/main/allennlp/training/learning_rate_schedulers/learning_rate_scheduler.py#L100) need metrics to make decisions.
The code in `_try_train` tried to fit all scheduler, hope that could help.

**Comment by bratao** [comment]
@wlhgtc yeah, you right. I use ReduceOnPlateauLearningRateScheduler , and in the latest version it was not detecting sn improvement because this_epoch_val_metric was stuck on 0

**Comment by dirkgr** [comment]
> I guess @dirkgr would solve it more elegantly as he was the designer of the improved metric tracker.

Haha, I love the confidence, but maybe I just screwed up. Let me take a look and understand this :-)

**Comment by dirkgr** [comment]
Fixed in #4959.
